tfhe/src/integer/server_key/radix/sub.rs

```diff
@@ -136,19 +137,32 @@ impl ServerKey {
             // and we also want to be able to add the carry from preceding block addition
             // to make sure carry propagation would be correct.
             if (degree_after_add + preceding_block_carry) >= total_modulus {
-                return false;
+                return Err(CheckError::CarryFull {
+                    degree: Degree(degree_after_add + preceding_block_carry),
+                    max_degree: MaxDegree(total_modulus - 1),
+                });
             }
 
+            let extracted_carry_noise_level = if first_block {
+                NoiseLevel::ZERO
+            } else {
+                NoiseLevel::NOMINAL
+            };
```

Comment:
same as for addition, init `extracted_carry_noise_level` to zero before the loop then assign Nominal to it

Reply:
Done